### PR TITLE
Add remote debugger port option to concourse.prefs

### PR DESCRIPTION
**Pull request by asyncRobert**
CON-577

Add remote debugger port option to Concourse prefs.

### CODE REVIEW

File: concourse-server/scripts/concourse [hunk 1]
```diff
@@ -57,7 +57,7 @@ JVMOPTS="
 -Dcom.sun.management.jmxremote.authenticate=false
 -Dcom.sun.management.jmxremote.ssl=false
 -XX:+UseThreadPriorities
--XX:ThreadPriorityPolicy=42
+-XX:ThreadPriorityPolicy=1
```
Comment (jtnelson) [inline concourse-server/scripts/concourse]: @asyncRobert Can you remove this change from this branch since it relates to another ticket.

Comment (asyncRobert) [reply concourse-server/scripts/concourse]: Oops, reverted.

File: concourse-server/conf/concourse.prefs [hunk 2]
```diff
@@ -148,3 +148,7 @@
 #
 # DEFAULT: automatically chosen based on the client_port
 #shutdown_port = 3434
+
+# The remote debugger port.
```
Comment (jtnelson) [inline concourse-server/conf/concourse.prefs]: @asyncRobert Set this as the pref description

```
# The listener port (1-65535) for remote debugger connections. Choose a port between
# 49152 and 65535 to minimize the possibility of conflicts with other services
# on this host. If the value of this preference is set to 0, then remote debugging for
# Concourse Server is disabled.
```

File: concourse-server/conf/concourse.prefs [hunk 3]
```diff
@@ -148,3 +148,7 @@
...
+# DEFAULT: 54555
```
Comment (jtnelson) [inline concourse-server/conf/concourse.prefs]: @asyncRobert Set the default to `0`

File: concourse-server/scripts/concourse [hunk 4]
```diff
@@ -233,6 +233,12 @@ if [ -z "$JMX_PREF" ]; then
 fi
 JVMOPTS=${JVMOPTS/jmxremote.port=9010/jmxremote.port=$JMX_PREF}
 
+#remote_debugger_port
+RDP_PREF=`cat $PREFS | grep -e '^remote_debugger_port\s*=\s*[0-9]\{1,\}$' | head -n1 | cut -d'=' -f2 | tr -d ' '`
+if [ -n "$RDP_PREF" ]; then
```
Comment (jtnelson) [inline concourse-server/scripts/concourse]: @asyncRobert Add a check to verify that the RDP_PREF value is not equal to `0`. If the pref is equal to 0, you should treat it as if the RDP_PREF is not provided.

File: concourse-server/scripts/concourse [hunk 5]
```diff
@@ -233,6 +233,12 @@ if [ -z "$JMX_PREF" ]; then
...
+  JVMOPTS=${JVMOPTS/-Xdebug/-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=$RDP_PREF}
```
Comment (jtnelson) [inline concourse-server/scripts/concourse]: @asyncRobert This logic is actually doing a find/replace which won't work in this case (sorry if I misled you in the original ticket)

You actually want to add the remote debugging flags to the JVMOPTS by doing
```
if [ -n "$RDP_PREF" ]; then
  JVMOPTS=$JVMOPTS"
  -Xdebug
  -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=$RDP_PREF
  "
fi
```

File: concourse-server/scripts/concourse [hunk 6]
```diff
@@ -233,6 +233,15 @@ if [ -z "$JMX_PREF" ]; then
 fi
 JVMOPTS=${JVMOPTS/jmxremote.port=9010/jmxremote.port=$JMX_PREF}
 
+#remote_debugger_port
+RDP_PREF=`cat $PREFS | grep -e '^remote_debugger_port\s*=\s*[0-9]\{1,\}$' | head -n1 | cut -d'=' -f2 | tr -d ' '`
+if [ "$RDP_PREF" -ne "0" ]; then
```
Comment (jtnelson) [inline concourse-server/scripts/concourse]: @asyncRobert What happens if RDP_PREF is empty. Would this check pass and the remote debugger flags would be added?

Comment (asyncRobert) [reply concourse-server/scripts/concourse]: I tested alternative logic that works with both scenarios and pushed to CON-577, let me know what you think!

File: concourse-server/scripts/concourse [hunk 7]
```diff
@@ -233,6 +233,17 @@ if [ -z "$JMX_PREF" ]; then
 fi
 JVMOPTS=${JVMOPTS/jmxremote.port=9010/jmxremote.port=$JMX_PREF}
 
+#remote_debugger_port
+RDP_PREF=`cat $PREFS | grep -e '^remote_debugger_port\s*=\s*[0-9]\{1,\}$' | head -n1 | cut -d'=' -f2 | tr -d ' '`
+if [ ! -z "$RDP_PREF" ]; then
+     if [ "$RDP_PREF" -ne "0" ]; then
```
Comment (jtnelson) [inline concourse-server/scripts/concourse]: @asyncRobert you should write this as one conditional similar to https://stackoverflow.com/a/13408590.

Comment (asyncRobert) [reply concourse-server/scripts/concourse]: pushed one conditional statement

**Pull request closed without merging**